File: src/main/java/me/cephetir/skyskipped/features/Features.java
/*
 *
 * DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE
 * Version 2, December 2004
 *
 * Copyright (C) 2021 Cephetir
 *
 * Everyone is permitted to copy and distribute verbatim or modified
 * copies of this license document, and changing it is allowed as long
 * as the name is changed.
 *
 * DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE
 * TERMS AND CONDITIONS FOR COPYING, DISTRIBUTION AND MODIFICATION
 *
 *  0. You just DO WHAT THE FUCK YOU WANT TO.
 */

package me.cephetir.skyskipped.features;

import lombok.Getter;
import me.cephetir.skyskipped.commands.fragrun.FragRun;
import me.cephetir.skyskipped.commands.fragrun.LeaveCommand;
import me.cephetir.skyskipped.commands.fragrun.PartyCommand;
import me.cephetir.skyskipped.features.impl.LastCrit;
import me.cephetir.skyskipped.features.impl.chat.ChatSwapper;
import me.cephetir.skyskipped.features.impl.chat.Nons;
import me.cephetir.skyskipped.features.impl.chat.Ping;
import me.cephetir.skyskipped.features.impl.dungeons.ChestCloser;
import me.cephetir.skyskipped.features.impl.dungeons.PizzaFailSafe;
import me.cephetir.skyskipped.features.impl.dungeons.PlayerESP;
import me.cephetir.skyskipped.features.impl.dungeons.ScoreCalculation;
import me.cephetir.skyskipped.features.impl.hacks.Blocker;
import me.cephetir.skyskipped.features.impl.visual.HidePetCandies;
import me.cephetir.skyskipped.features.impl.visual.PetsOverlay;
import net.minecraftforge.common.MinecraftForge;

import java.util.ArrayList;
import java.util.List;

public class Features {
    @Getter
    private final LeaveCommand leaveCommand = new LeaveCommand();
    @Getter
    private final PartyCommand partyCommand = new PartyCommand();
    @Getter
    private final ScoreCalculation scoreCalculation = new ScoreCalculation();
    @Getter
    private final PetsOverlay petsOverlay = new PetsOverlay();

    public List<Feature> features = new ArrayList<>();

    public void register() {
        features.add(new ChestCloser());
        features.add(new ChatSwapper());
        features.add(new PlayerESP());
        features.add(new LastCrit());
        features.add(new Nons());
        features.add(new Blocker());
        features.add(new Ping());
        features.add(new FragRun());
        features.add(getScoreCalculation());
        features.add(new PizzaFailSafe());
        features.add(new HidePetCandies());
        features.add(getPetsOverlay());
        features.forEach(MinecraftForge.EVENT_BUS::register);
    }
}

File: src/main/kotlin/me/cephetir/skyskipped/features/Features.kt
/*
 * SkySkipped - Hypixel Skyblock QOL mod
 * Copyright (C) 2023  Cephetir
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package me.cephetir.skyskipped.features

import me.cephetir.bladecore.core.event.BladeEventBus
import me.cephetir.skyskipped.commands.dungeonCommands.FragRun
import me.cephetir.skyskipped.commands.dungeonCommands.LeaveCommand
import me.cephetir.skyskipped.commands.dungeonCommands.PartyCommand
import me.cephetir.skyskipped.features.impl.chat.*
import me.cephetir.skyskipped.features.impl.cofl.AutoFuckingCoflBuy
import me.cephetir.skyskipped.features.impl.discordrpc.RPC
import me.cephetir.skyskipped.features.impl.dugeons.*
import me.cephetir.skyskipped.features.impl.hacks.*
import me.cephetir.skyskipped.features.impl.macro.FarmingHUD
import me.cephetir.skyskipped.features.impl.macro.GuiRecorder
import me.cephetir.skyskipped.features.impl.macro.MacroManager
import me.cephetir.skyskipped.features.impl.misc.*
import me.cephetir.skyskipped.features.impl.movement.AutoStopFlying
import me.cephetir.skyskipped.features.impl.optimization.HideDamageInBoss
import me.cephetir.skyskipped.features.impl.optimization.HideWitherCloak
import me.cephetir.skyskipped.features.impl.optimization.StopFallingBlocks
import me.cephetir.skyskipped.features.impl.visual.*

object Features {
    val leaveCommand = LeaveCommand()
    val partyCommand = PartyCommand()
    val petsOverlay = PetsOverlay()

    private val features = listOf(
        ChestCloser(),
        ChatSwapper(),
        RPC,
        ESP(),
        Blocker(),
        Ping(),
        FragRun(),
        Pings(),
        HidePetCandies(),
        petsOverlay,
        PresentHighlight(),
        GhostBlocks,
        PerspectiveToggle(),
        AutoMaddoxPhone(),
        CustomScoreboard(),
        HighlightUnlockedGemSots(),
        ItemSwap(),
        AutoStopFlying(),
        Metrics(),
        AutoCookieClicker(),
        Trail(),
        MacroManager,
        FarmingHUD(),
        HotbarSaver,
        LavaFishingSpots(),
        ItemRadius(),
        TerminalEsp(),
        AntiEscrow(),
        AutoSalvage(),
        ChatSearch(),
        SearchContainer(),
        AutoReply(),
        AutoRedirectClick(),
        ShinyBlocks(),
        StopFallingBlocks(),
        HideDamageInBoss(),
        DungeonScan,
        JerryBoxOpener(),
        BanDetector(),
        TerminatorClicker(),
        AOTVDisplay(),
        Garden(),
        HideWitherCloak(),
        GuiRecorder,
        BlessingsDisplay(),
        TokenAuth,
        Harp(),
        AutoFuckingCoflBuy,
        ChatHider(),
        ZeroPingEtherwarp(),
        FishingAssist(),
        Rift(),
        AutoAttribPickup(),
        NoCursorReset,
    )

    fun register() = features.forEach { BladeEventBus.subscribe(it, true) }
}
